lib/SILGen/SILGenConvert.cpp

```diff
@@ -856,24 +856,20 @@ SILGenFunction::emitOpenExistential(
                "value mode");
         loweredOpenedType = loweredOpenedType.getAddressType();
       }
-      archetypeValue =
+      SILValue archetypeValue =
           B.createOpenExistentialAddr(loc, existentialValue.forward(*this),
                                       loweredOpenedType, allowedAccess);
-    } else {
-      archetypeValue = B.createOpenExistentialValue(
-          loc, existentialValue.forward(*this), loweredOpenedType);
-    }
-
-    if (existentialValue.hasCleanup()) {
-      // With CoW existentials we can't consume the boxed value inside of
-      // the existential. (We could only do so after a uniqueness check on
-      // the box holding the value).
-      canConsume = false;
-      enterDestroyCleanup(existentialValue.getValue());
+      if (existentialValue.hasCleanup()) {
```

Comment:
That's a different code path than what I was testing. I assumed that code was written in a convoluted fashion for some good reason. I'll try to clean it up and see what breaks.